lib/pbench/server/api/resources/upload_api.py

```diff
@@ -48,10 +49,14 @@ def __init__(self, config, logger):
             )
         )
 
+    @Auth.token_auth.login_required()
     def put(self, controller):
 
-        # FIXME: This should be assigned from the decoded authorization token
-        username = "pbench"
+        try:
+            username = Auth.token_auth.current_user().username
+        except Exception:
+            self.logger.exception("Tarfile upload: Exception verifying the username")
+            abort(500, message="INTERNAL ERROR")
```

Comment:
You might as well consider using `HTTPStatus.INTERNAL_SERVER_ERROR` here if you come back to do any further edits.